masonry/src/widgets/slider.rs

```diff
@@ -0,0 +1,393 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use tracing::{Span, trace_span};
+use vello::Scene;
+use vello::kurbo::{Affine, Circle, Point, Rect, Size, Stroke};
+use vello::peniko::{Brush, Color, Fill};
+
+use crate::accesskit::{Node, Role};
+use crate::core::{
+    AccessCtx, AccessEvent, BoxConstraints, ChildrenIds, EventCtx, LayoutCtx, PaintCtx,
+    PointerEvent, PropertiesMut, PropertiesRef, RegisterCtx, TextEvent, Update, UpdateCtx, Widget,
+    WidgetId, WidgetMut,
+};
+use crate::theme;
+
+/// A widget that allows a user to select a value from a continuous range.
+pub struct Slider {
+    min: f64,
+    max: f64,
+    value: f64,
+    step: Option<f64>,
+    is_dragging: bool,
+    disabled: bool,
+    track_color: Option<Color>,
+    active_track_color: Option<Color>,
+    thumb_color: Option<Color>,
+    track_thickness: Option<f64>,
+    thumb_radius: Option<f64>,
```

Comment:
Some of these need sorting to have track properties next to track properties and thumb properties with thumb properties. Sort the methods in similar ways as well.